src/Core/Silk.NET.Core.Tests/TestSilkMarshal.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Silk.NET.Core.Native;
+using Xunit;
+
+namespace Silk.NET.Core.Tests;
+
+public class TestSilkMarshal
+{
+    private readonly List<NativeStringEncoding> encodings = new()
+    {
+        NativeStringEncoding.BStr,
+        NativeStringEncoding.LPStr,
+        NativeStringEncoding.LPTStr,
+        NativeStringEncoding.LPUTF8Str,
+        NativeStringEncoding.LPWStr,
+    };
+
+    [Fact]
+    public unsafe void TestEncodingToLPWStr()
+    {
+        var input = "Hello world";
+        
+        // LPWStr is 2 bytes on Windows, 4 bytes elsewhere (usually)
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            var pointer = SilkMarshal.StringToPtr(input, NativeStringEncoding.LPWStr);
+            
+            Assert.Equal(input.Length, (int)SilkMarshal.StringLength(pointer, NativeStringEncoding.LPWStr));
+            
+            // Use short for comparison
+            Assert.Equal(new short[] { 0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x77, 0x6f, 0x72, 0x6c, 0x64, 0x00 }, new Span<short>((void*)pointer, input.Length + 1));
+        }
+        else
+        {
+            var pointer = SilkMarshal.StringToPtr(input, NativeStringEncoding.LPWStr);
+            
+            Assert.Equal(input.Length, (int)SilkMarshal.StringLength(pointer, NativeStringEncoding.LPWStr));
+            
+            // Use int for comparison
+            Assert.Equal(new int[] { 0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x77, 0x6f, 0x72, 0x6c, 0x64, 0x00 }, new Span<int>((void*)pointer, input.Length + 1));
+        }
+    }
+    
+    [Fact]
+    public unsafe void TestEncodingFromLPWStr()
+    {
+        var expected = "Hello world";
+        
+        // LPWStr is 2 bytes on Windows, 4 bytes elsewhere (usually)
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            var characters = new short[] { 0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x77, 0x6f, 0x72, 0x6c, 0x64 };
```

Comment:
Why not just have a field with `RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? Encoding.Unicode : Encoding.UTF32` and call GetBytes on it here?

Reply:
That's doable and I did miss this when I was updating the code to use Encoding.UTF32.
I am hesitant to do this though because I'm unfamiliar with the exact differences between UTF16 and UTF32.

The difference in this case is that all of the 2-byte UTF16 characters are widened into 4-byte characters by inserting zeroes (due to the size difference of wchar_t between Windows and non-Windows).

Is this how the conversion between UTF16 and UTF32 is done normally?